File: android/app/src/main/java/com/adasa/foodplan/domain/model/RecipeRating.kt
package com.adasa.foodplan.domain.model

import kotlinx.datetime.LocalDate

data class RecipeRating(
    val recipeId: String,
    val stars: Int? = null,                    // 1–5, null = unrated
    val timesScheduled: Int = 0,
    val timesManuallyRemoved: Int = 0,
    val isPinned: Boolean = false,
    val isExcluded: Boolean = false,
    val lastScheduledDate: LocalDate? = null
)

File: flutter-app/lib/domain/model/recipe_rating.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'recipe_rating.freezed.dart';

@freezed
class RecipeRating with _$RecipeRating {
  const factory RecipeRating({
    required String recipeId,
    int? stars, // 1–5, null = unrated
    @Default(0) int timesScheduled,
    @Default(0) int timesManuallyRemoved,
    @Default(false) bool isPinned,
    @Default(false) bool isExcluded,
    DateTime? lastScheduledDate,
  }) = _RecipeRating;
}
